test/helpers/errors.go

```diff
@@ -0,0 +1,29 @@
+package helpers
+
+import "fmt"
+
+// NewSSHMetaError returns a new SSHMetaError with the given string and a
+// callback function
+func NewSSHMetaError(s string, callback func() string) *SSHMetaError {
+	return &SSHMetaError{
+		s:       s,
+		errorCB: callback,
```

Comment:
Check if the provided callback is nil; If it is nil, put a dummy-callback in place to ensure that no panic occurs when `String()` is called, which unconditionally invokes the function. Or, you can ensure that errorCB is not nil in functions which utilize it.